Fix #150, #151, #155: TextInputInteractionView, NumericInputInteractionView, FractionInputInteractionView

nikitamarysolomanpvt opened the pull request:
## Explanation
This PR contains Low-fi TextInputInteractionView, NumberInputInteractionView, FractionInputInteractionView with dummy call implementation in StateFragmentPresenter. #150, #151, #155.
## Reference Mocks
https://xd.adobe.com/spec/e2239cf4-9cde-4c08-5296-25316c1f0a14-9412/screen/4d4e4bd8-3755-41fc-bf83-3dcff5d5eee1/Lesson-1-a-ii-Exploration-Player-Text-Input-No-Ans
https://xd.adobe.com/spec/e2239cf4-9cde-4c08-5296-25316c1f0a14-9412/screen/eeed0f6c-54ca-4371-ab23-5199abd87f28/Lesson-1-a-ii-Exploration-Player-Text-Input-Keyboa
https://xd.adobe.com/spec/e2239cf4-9cde-4c08-5296-25316c1f0a14-9412/screen/d66097bf-8421-41d8-bfda-2450d6e98dea/Lesson-1-a-i-Exploration-Player-Fraction-Input-No-
https://xd.adobe.com/spec/e2239cf4-9cde-4c08-5296-25316c1f0a14-9412/screen/249c3349-605b-4569-bbe9-1a017ab90101/Lesson-1-a-i-Exploration-Player-Fraction-Input-No-
https://xd.adobe.com/spec/e2239cf4-9cde-4c08-5296-25316c1f0a14-9412/screen/c4b0dfb3-53b4-491e-a759-818e096183fd/Lesson-1-a-v-Exploration-Player-Numeric-Input-No-A

## Reference Design Doc
https://docs.google.com/document/d/1QigNdZeMBy-JjHW26QH4qT0UtCyzfvW0xEaStSdY9gM/edit?usp=sharing

Commit:
Exploration Player base code for activity and fragment

Commit:
Fake Data Provider

Commit:
Merge remote-tracking branch 'upstream/develop' into HEAD

Commit:
Nit changes mainly in documentation

Commit:
Nit changes mainly in documentation for State too

Commit:
Changed Controller to Presenter and optimised the fragment transaction code

Commit:
Basic code for ExporationActivityTest

Commit:
ExplorationActivity dummy test

Commit:
Merge remote-tracking branch 'upstream/develop' into exploration-player-1-base

Commit:
Nit changes

Commit:
Merge remote-tracking branch 'upstream/develop' into exploration-player-1-base

Commit:
Code checking and refactoring in ExplorationActivityTest

nikitamarysolomanpvt committed:
codeStyle

nikitamarysolomanpvt committed:
TextInputInteractionView

nikitamarysolomanpvt committed:
TextInputInteractionView

nikitamarysolomanpvt committed:
TextInputInteractionView ,NumberInputInteractionView,FractionInputInteractionView

nikitamarysolomanpvt committed:
TextInputInteractionView,NumberInputInteractionView,FractionInputInteractionView
Kdoc for inputinteractioncustomviews
added color resourses for edittext
Merge branches 'develop' and 'exploration-player-3-interaction' of https://github.com/oppia/oppia-android into exploration-player-3-interaction

# Conflicts:
#	app/src/main/java/org/oppia/app/fragment/FragmentComponent.kt
#	app/src/main/java/org/oppia/app/player/state/StateFragment.kt
#	app/src/main/java/org/oppia/app/player/state/StateFragmentPresenter.kt
#	app/src/main/res/layout/exploration_fragment.xml
#	app/src/main/res/layout/state_fragment.xml
#	app/src/main/res/values/dimens.xml
#	app/src/sharedTest/java/org/oppia/app/player/exploration/ExplorationActivityTest.kt

rt4914 commented on .idea/codeStyles/codeStyleConfig.xml:
```diff
@@ -1,5 +1,6 @@
+    <option name="PREFERRED_PROJECT_CODE_STYLE" value="Project copy" />
   </state>
-</component>
+</component>
```
Remove this file from your commit, you can select/deselect which files you want to commit in android studio.

nikitamarysolomanpvt replied:
okay

rt4914 replied:
You have marked it as resolved but it has not been resolved.

Also, if you are not able to remove this file, then maybe correct this file by adding EOF

nikitamarysolomanpvt replied:
okay

rt4914 commented on app/build.gradle:
```diff
@@ -37,7 +37,9 @@ android {
     execution 'ANDROIDX_TEST_ORCHESTRATOR'
     unitTests {
       includeAndroidResources = true
+
```
Remove this extra line.

nikitamarysolomanpvt replied:
resolved

rt4914 replied:
You have marked this resolved but this has not been resolved.

nikitamarysolomanpvt replied:
the changes were not pushed successfully

rt4914 commented on app/src/main/java/org/oppia/app/customview/inputInteractionView/FractionInputInteractionView.kt:
```diff
@@ -0,0 +1,42 @@
+  }
+
+  //update default attributes of TextInputInteractionView here
+  fun init() {
```
Kotlin has a concept of Primary and Secondary Constructor and `init` is a type of secondary constructor and its not good to give a function name `init` because of secondary constructor name. Also, I think you can use the concept of primary and secondary constructor here for clearer code.

Reference: https://blog.mindorks.com/primary-and-secondary-constructors-in-kotlin

nikitamarysolomanpvt replied:
resolved

rt4914 commented on app/src/main/java/org/oppia/app/customview/inputInteractionView/FractionInputInteractionView.kt:
```diff
@@ -0,0 +1,42 @@
+    init()
+  }
+
+  //update default attributes of TextInputInteractionView here
```
Write comments properly.
Example: `// This function contains default attributes of TextInputInteractionView. `

rt4914 replied:
Nit: After `//` there should be one space, first character should be capital and before end of sentence there should be `.`

nikitamarysolomanpvt replied:
okay

rt4914 commented on app/src/main/java/org/oppia/app/customview/inputInteractionView/FractionInputInteractionView.kt:
```diff
@@ -0,0 +1,42 @@
+    filterArray[0] = InputFilter.LengthFilter(length)
+    setFilters(filterArray)
+  }
+
```
Remove extra space.

nikitamarysolomanpvt replied:
okay

rt4914 commented on app/src/main/java/org/oppia/app/customview/inputInteractionView/FractionInputInteractionView.kt:
```diff
@@ -0,0 +1,42 @@
+  }
+
+}
+
```
Remove extra space.

nikitamarysolomanpvt replied:
resolved

rt4914 commented on app/src/main/java/org/oppia/app/customview/inputInteractionView/NumberInputInteractionView.kt:
```diff
@@ -0,0 +1,44 @@
+  }
+
+}
+
```
Apply changes similar to `FractionInputInteractionView`.

nikitamarysolomanpvt replied:
resolved

rt4914 commented on app/src/main/java/org/oppia/app/customview/inputInteractionView/TextInputInteractionView.kt:
```diff
@@ -0,0 +1,43 @@
+
+
+}
+
```
Apply changes similar to `FractionInputInteractionView`.

nikitamarysolomanpvt replied:
resolved

rt4914 commented on app/src/main/java/org/oppia/app/fragment/FragmentComponent.kt:
```diff
@@ -19,8 +19,8 @@ interface FragmentComponent {
 
-  fun inject(audioFragment: AudioFragment)
   fun inject(explorationFragment: ExplorationFragment)
+  fun inject(audioFragment: AudioFragment)
```
I think there are no changes in this file, in such cases make sure that you revert changes in such files, so as to make review process faster.

rt4914 replied:
This conversation has been marked as resolved but has not been solved yet.

nikitamarysolomanpvt replied:
This would happen with several files due to merge, need to have verbal discussion on this .As if now iam reverting this file

nikitamarysolomanpvt replied:
copied and develop code pasted the code as revert was not helping

rt4914 commented on app/src/main/java/org/oppia/app/home/HomeActivityController.kt:
```diff
@@ -12,7 +13,7 @@ class HomeActivityController @Inject constructor(private val activity: AppCompat
     activity.supportFragmentManager.beginTransaction().add(
       R.id.home_fragment_placeholder,
-      HomeFragment()
+      StateFragment()
```
Revert this to `HomeFragment()` itself, so that this PR is ready for merge. By now in all PRs it is understood that we will have to change this to `StateFragment` to test your code. Also, better you write code which is merge-ready and in that if there are some steps which needs to be performed to check this PR mention that in the description of PR.

nikitamarysolomanpvt replied:
okay

rt4914 commented on app/src/main/java/org/oppia/app/home/HomeFragment.kt:
```diff
@@ -17,7 +19,9 @@ class HomeFragment : InjectableFragment() {
   override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
     return homeFragmentController.handleCreateView(inflater, container)
   }
+
```
Remove extra space

nikitamarysolomanpvt replied:
resolved

rt4914 commented on app/src/main/java/org/oppia/app/home/HomeFragmentController.kt:
```diff
@@ -17,20 +22,47 @@ class HomeFragmentController @Inject constructor(
-
+    llRoot = binding.root.findViewById(R.id.llRoot)
+    addContentCard("edit_text_background_border",1)
+    addContentCard("edit_text_background_border_",2)
```
There are no spaces after `,`.
Correct:
`addContentCard("edit_text_background_border_", 2)`

nikitamarysolomanpvt replied:
resolved

rt4914 commented on app/src/main/java/org/oppia/app/home/HomeFragmentController.kt:
```diff
@@ -17,20 +22,47 @@ class HomeFragmentController @Inject constructor(
+    llRoot!!.addView(contentComponent, params)
+
+
+
```
Please make sure your PR does not have unnecessary space.

nikitamarysolomanpvt replied:
resolved

rt4914 commented on app/src/main/java/org/oppia/app/home/HomeFragmentController.kt:
```diff
@@ -17,20 +22,47 @@ class HomeFragmentController @Inject constructor(
+      LinearLayout.LayoutParams.WRAP_CONTENT
+    )
+
+    params.setMargins( dpToPx(8), dpToPx(8),  dpToPx(8), dpToPx(8))
```
Correct
`params.setMargins(dpToPx(8), dpToPx(8), dpToPx(8), dpToPx(8))`

Also, better if define `8` as a constant.

nikitamarysolomanpvt replied:
okay

rt4914 commented on app/src/main/java/org/oppia/app/player/state/StateFragmentPresenter.kt:
```diff
@@ -15,15 +21,73 @@ class StateFragmentPresenter @Inject constructor(
+    )
+    params.setMargins(dpToPx(8), dpToPx(8), dpToPx(8), dpToPx(8))
+    llRoot!!.addView(contentComponent, params)
+
```
Remove this blank line

nikitamarysolomanpvt replied:
resolved

rt4914 commented on app/src/main/java/org/oppia/app/player/state/StateFragmentPresenter.kt:
```diff
@@ -15,15 +21,73 @@ class StateFragmentPresenter @Inject constructor(
+
+  }
+
+  //for adding numberinputinteraction view
```
Ditto: Comments as per my above description

nikitamarysolomanpvt replied:
resolved

rt4914 commented on app/src/main/java/org/oppia/app/player/state/StateFragmentPresenter.kt:
```diff
@@ -15,15 +21,73 @@ class StateFragmentPresenter @Inject constructor(
+
+  }
+
+  //for adding numberinputinteraction view
```
Ditto: Comments as per my above description

rt4914 commented on app/src/main/res/drawable/edit_text_background.xml:
```diff
@@ -0,0 +1,7 @@
+  <item android:drawable="@drawable/edit_text_background_shadow" android:state_focused="true" />
+  <item android:drawable="@drawable/edit_text_background_shadow" android:state_pressed="true" />
+  <item android:drawable="@drawable/edit_text_background_border" />
+  //means normal
```
Remove this comment

nikitamarysolomanpvt replied:
shifted it to right of tag for understanding normal tag

rt4914 commented on app/src/main/res/drawable/edit_text_background_border.xml:
```diff
@@ -0,0 +1,9 @@
+  <corners android:radius="3dp" />
+  <solid android:color="@color/white" />
+  <stroke
+    android:width="1dip"
```
Do we need this as `dip`?

nikitamarysolomanpvt replied:
resolved

rt4914 commented on app/src/main/res/drawable/edit_text_background_border.xml:
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android"
+  android:shape="rectangle">
+  <corners android:radius="3dp" />
```
Check mocks -> I think it should be `4dp`

nikitamarysolomanpvt replied:
okay

rt4914 commented on app/src/main/res/drawable/edit_text_background_shadow.xml:
```diff
@@ -0,0 +1,64 @@
+  </item>
+  <item>
+    <shape>
+      <corners android:radius="5dp"></corners>
```
Is this `5dp` correct

nikitamarysolomanpvt replied:
ya it is

rt4914 replied:
If you are checking that from mock, then also make it `4dp`, @BenHenning has already communicated with Chantel regarding this  and everything needs to be in multiple of 4 and 8

rt4914 commented on app/src/main/res/layout/home_fragment.xml:
```diff
@@ -10,19 +11,24 @@
+<!--      android:layout_margin="20dp"-->
+<!--      app:layout_constraintTop_toTopOf="parent"/>-->
+
+  </LinearLayout>
```
You don't need to make changes to `home_fragment` file for this PR.

nikitamarysolomanpvt replied:
had reverted the  changes

rt4914 commented on app/src/main/res/values/dimens.xml:
```diff
@@ -1,5 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources>
+  <dimen name="radius_corner">5</dimen>
```
`5` or `5dp`

nikitamarysolomanpvt replied:
removed

rt4914 commented on app/src/sharedTest/java/org/oppia/app/home/HomeActivityTest.kt:
```diff
@@ -74,7 +74,7 @@ class HomeActivityTest {
+//      onView(withId(R.id.welcome_text_view)).check(matches(withText("Welcome back to Oppia!")))
     }
   }
 
```
Why do you need to make changes to `HomeActivityTest` ?

nikitamarysolomanpvt replied:
initially i created dummy view call and implementation in this , now reverted the code

rt4914 commented on app/src/sharedTest/java/org/oppia/app/player/state/StateFragmentPresenterTest.kt:
```diff
@@ -0,0 +1,97 @@
+    ActivityScenario.launch(HomeActivity::class.java).use {
+      onView(ViewMatchers.withId(R.id.dummy_audio_button)).perform(click())
+      onView(ViewMatchers.withHint("text"))
+        .check(matches(isDisplayed()))
```
Fix this in one single line for faster review process

nikitamarysolomanpvt replied:
okay

rt4914 commented on app/src/sharedTest/java/org/oppia/app/player/state/StateFragmentPresenterTest.kt:
```diff
@@ -0,0 +1,97 @@
+
+@RunWith(AndroidJUnit4::class)
+class StateFragmentPresenterTest {
+  //to test text,number,and fraction input interaction views
```
Please add proper comments as per my description earlier.

nikitamarysolomanpvt replied:
okay

rt4914 reviewed (commented):
PTAL

nikitamarysolomanpvt committed:
TextInputInteractionView,NumberInputInteractionView,FractionInputInteractionView
Kdoc for inputinteractioncustomviews
added color resourses for edittext
Merge branches 'develop' and 'exploration-player-3-interaction' of https://github.com/oppia/oppia-android into exploration-player-3-interaction

# Conflicts:
#	app/src/main/java/org/oppia/app/fragment/FragmentComponent.kt
#	app/src/main/java/org/oppia/app/player/state/StateFragment.kt
#	app/src/main/java/org/oppia/app/player/state/StateFragmentPresenter.kt
#	app/src/main/res/layout/exploration_fragment.xml
#	app/src/main/res/layout/state_fragment.xml
#	app/src/main/res/values/dimens.xml
#	app/src/sharedTest/java/org/oppia/app/player/exploration/ExplorationActivityTest.kt

rt4914 commented:
@nikitamarysolomanpvt Please update your PR title to point to issues.

rt4914 commented:
@nikitamarysolomanpvt You have marked some conversations as resolved but they have not been resolved. Please check the first and assign it back to me.

rt4914 reviewed (commented):
PTAL

nikitamarysolomanpvt committed:
PR #190 ressolved issues

nikitamarysolomanpvt committed:
PR #190 ressolved issues

rt4914 commented on app/src/main/java/org/oppia/app/customview/inputInteractionView/TextInputInteractionView.kt:
```diff
@@ -0,0 +1,41 @@
+    attributes()
+  }
+
+  // This function contains default attributes of TextInputInteractionView.
```
Update:

// This function contains default attributes of [TextInputInteractionView].

Here and everywhere else in PR.

nikitamarysolomanpvt replied:
resolved as
/** This function contains default attributes of [TextInputInteractionView].*/

rt4914 commented on app/src/main/java/org/oppia/app/home/HomeFragmentController.kt:
```diff
@@ -23,11 +28,9 @@ class HomeFragmentController @Inject constructor(
-
     return binding.root
   }
 
```
Remove this extra line

nikitamarysolomanpvt replied:
There is no extra line

rt4914 commented on app/src/main/java/org/oppia/app/player/state/StateFragmentPresenter.kt:
```diff
@@ -15,15 +21,72 @@ class StateFragmentPresenter @Inject constructor(
     return binding.root
   }
 
+  // The function for adding Textinputinteraction view
```
Ditto: Directly link TextInputInteractionView here

rt4914 replied:
Same for everywhere else in this PR

nikitamarysolomanpvt replied:
okay

rt4914 commented on app/src/main/java/org/oppia/app/player/state/StateFragmentPresenter.kt:
```diff
@@ -15,15 +21,72 @@ class StateFragmentPresenter @Inject constructor(
+    )
+    params.setMargins(dpToPx(8), dpToPx(8), dpToPx(8), dpToPx(8))
+    llRoot!!.addView(contentComponent, params)
+
```
Remove this extra line

nikitamarysolomanpvt replied:
resolved

rt4914 commented on app/src/main/res/drawable/edit_text_background_shadow.xml:
```diff
@@ -0,0 +1,64 @@
+<layer-list xmlns:android="http://schemas.android.com/apk/res/android">
+  <item>
+    <shape>
+      <corners android:radius="5dp"></corners>
```
The dimen should be in multiple of 4 and sometimes 2 is fine. Change this to `5dp`

nikitamarysolomanpvt replied:
This is not related to the view dimension, this attribute was for the layer of shadow customisation

rt4914 replied:
Got it. Thanks.

rt4914 commented on app/src/main/res/layout/home_fragment.xml:
```diff
@@ -1,7 +1,8 @@
   xmlns:android="http://schemas.android.com/apk/res/android"
-  xmlns:app="http://schemas.android.com/apk/res-auto">
+  xmlns:app="http://schemas.android.com/apk/res-auto"
+  android:orientation="vertical">
```
What does this `orientation` do here?

nikitamarysolomanpvt replied:
resolved

rt4914 commented on app/src/main/res/layout/state_fragment.xml:
```diff
@@ -15,19 +15,32 @@
       app:layout_constraintStart_toStartOf="parent"
-      app:layout_constraintTop_toTopOf="parent">
+      app:layout_constraintTop_toTopOf="parent"
+      android:orientation="vertical">
```
Alphabetical order, use auto-arrange shortcut on every file before PR

nikitamarysolomanpvt replied:
okay

rt4914 commented on app/src/main/res/layout/state_fragment.xml:
```diff
@@ -15,19 +15,32 @@
+      app:layout_constraintEnd_toEndOf="parent"
+      app:layout_constraintStart_toStartOf="parent"
+      app:layout_constraintTop_toTopOf="parent"
+      android:orientation="vertical"></LinearLayout>
```
Make this correct

nikitamarysolomanpvt replied:
resolved

rt4914 commented on app/src/main/res/values/styles.xml:
```diff
@@ -23,4 +23,5 @@
     <item name="android:windowEnterAnimation">@anim/slide_up</item>
     <item name="android:windowExitAnimation">@anim/slide_down</item>
   </style>
+
```
Remove this extra line

nikitamarysolomanpvt replied:
resolved

rt4914 reviewed (commented):
PTAL, also, if the reviewer, suggests some changes or asks as question and then if there is no change in that file/code, then do not resolve the conversation, instead try to answer why their is no change needed.

nikitamarysolomanpvt commented:
> @nikitamarysolomanpvt Please update your PR title to point to issues.

okay

nikitamarysolomanpvt committed:
PR #190 ressolved issues

nikitamarysolomanpvt committed:
PR #190 ressolved issues

nikitamarysolomanpvt committed:
PR #190 ressolved issues

nikitamarysolomanpvt committed:
PR #190 ressolved issues

nikitamarysolomanpvt committed:
PR #190 ressolved issues

nikitamarysolomanpvt committed:
PR #84 suggestion of ben fixes

nikitamarysolomanpvt commented:
> PTAL, also, if the reviewer, suggests some changes or asks as question and then if there is no change in that file/code, then do not resolve the conversation, instead try to answer why their is no change needed.

okay

nikitamarysolomanpvt commented:
> @nikitamarysolomanpvt Please update your PR title to point to issues.

resolved

nikitamarysolomanpvt committed:
Merge branches 'develop' and 'exploration-player-3-interaction' of https://github.com/oppia/oppia-android into exploration-player-3-interaction-input-type

# Conflicts:
#	.idea/codeStyles/codeStyleConfig.xml
#	app/src/sharedTest/java/org/oppia/app/player/exploration/ExplorationActivityTest.kt

rt4914 commented on app/src/main/java/org/oppia/app/customview/inputInteractionView/FractionInputInteractionView.kt:
```diff
@@ -0,0 +1,40 @@
+  /** This function contains default attributes of [FractionInputInteractionView].*/
+  fun attributes() {
+    setBackgroundResource(R.drawable.edit_text_background)
+    val paddingPixel = 10
```
Can this be 8? Assuming it refers to DP unit.

rt4914 commented on app/src/main/java/org/oppia/app/customview/inputInteractionView/FractionInputInteractionView.kt:
```diff
@@ -0,0 +1,40 @@
+  /** This function contains default attributes of [FractionInputInteractionView].*/
+  fun attributes() {
+    setBackgroundResource(R.drawable.edit_text_background)
+    val paddingPixel = 10
```
Can this be 8? Assuming it refers to DP unit.

nikitamarysolomanpvt replied:
resolved

rt4914 commented on app/src/main/java/org/oppia/app/customview/inputInteractionView/NumberInputInteractionView.kt:
```diff
@@ -0,0 +1,40 @@
+    attributes()
+  }
+
+  /** This function contains default attributes of [NumberInputInteractionView].*/
```
Nit: Add space after `.`

nikitamarysolomanpvt replied:
resolved

rt4914 commented on app/src/main/java/org/oppia/app/customview/inputInteractionView/TextInputInteractionView.kt:
```diff
@@ -0,0 +1,38 @@
+    attributes()
+  }
+
+  /** This function contains default attributes of [TextInputInteractionView].*/
```
Ditto.

rt4914 commented on app/src/main/java/org/oppia/app/customview/inputInteractionView/TextInputInteractionView.kt:
```diff
@@ -0,0 +1,38 @@
+  /** This function contains default attributes of [TextInputInteractionView].*/
+  fun attributes() {
+    setBackgroundResource(R.drawable.edit_text_background)
+    val paddingPixel = 10
```
Ditto

nikitamarysolomanpvt replied:
resolved

rt4914 commented on app/src/main/res/drawable/edit_text_background.xml:
```diff
@@ -0,0 +1,6 @@
+<selector xmlns:android="http://schemas.android.com/apk/res/android">
+  <item android:drawable="@drawable/edit_text_background_shadow" android:state_focused="true" />
+  <item android:drawable="@drawable/edit_text_background_shadow" android:state_pressed="true" />
+  <item android:drawable="@drawable/edit_text_background_border" />//means normal
```
Remove this comment. We don't need this. Also, comments in XML are of different format.
Example: `<!-- means normal -->`

nikitamarysolomanpvt replied:
resolved

rt4914 commented on app/src/main/res/layout/audio_fragment.xml:
```diff
@@ -1,14 +1,18 @@
     <variable
       name="viewModel"
-      type="org.oppia.app.player.audio.AudioViewModel"/>
+      type="org.oppia.app.player.audio.AudioViewModel" />
```
Remove this extra space at the end.

rt4914 replied:
Ditto elsewhere

rt4914 commented on app/src/main/res/layout/audio_fragment.xml:
```diff
@@ -1,14 +1,18 @@
     <variable
       name="audioFragment"
-      type="org.oppia.app.player.audio.AudioFragment"/>
+      type="org.oppia.app.player.audio.AudioFragment" />
```
Remove this extra space at the end.

nikitamarysolomanpvt replied:
resolved

rt4914 commented on app/src/main/res/layout/cellular_data_dialog.xml:
```diff
@@ -1,14 +1,13 @@
     android:orientation="vertical"
-    android:text="@string/cellular_data_alert_dialog_checkbox">
-  </CheckBox>
+    android:text="@string/cellular_data_alert_dialog_checkbox"></CheckBox>
```
End with self-closing tag. `/>`

nikitamarysolomanpvt replied:
reverting changes to develop code

nikitamarysolomanpvt replied:
resolved

rt4914 commented on app/src/main/res/layout/conceptcard_fragment.xml:
```diff
@@ -1,40 +1,41 @@
     <variable
       name="viewModel"
-      type="org.oppia.app.topic.conceptcard.ConceptCardViewModel"/>
+      type="org.oppia.app.topic.conceptcard.ConceptCardViewModel" />
```
Ditto

nikitamarysolomanpvt replied:
resolved

rt4914 commented on app/src/main/res/layout/home_fragment.xml:
```diff
@@ -22,7 +23,6 @@
-
+      app:layout_constraintTop_toTopOf="parent" />
   </androidx.constraintlayout.widget.ConstraintLayout>
 </layout>
```
If you have not done any changes which affect your code, then undo all these changes. Also, follow this for other files too.

nikitamarysolomanpvt replied:
resolved

rt4914 commented on app/src/main/res/values/styles.xml:
```diff
@@ -10,15 +10,11 @@
-
     <!-- Additionally if you want animations when dialog opening -->
     <item name="android:windowEnterAnimation">@anim/slide_up</item>
     <item name="android:windowExitAnimation">@anim/slide_down</item>
```
Ditto: revert this file.

nikitamarysolomanpvt replied:
resolved

rt4914 commented on app/src/sharedTest/java/org/oppia/app/player/state/StateFragmentPresenterTest.kt:
```diff
@@ -0,0 +1,88 @@
+  @Test
+  fun addTextInputContentCard() {
+    ActivityScenario.launch(HomeActivity::class.java).use {
+      onView(ViewMatchers.withId(R.id.dummy_audio_button)).perform(click())
```
Import `ViewMathers.withId` and `ViewMatchers.withHint` globally to make this code more readable

nikitamarysolomanpvt replied:
okay, resolved

rt4914 reviewed (commented):
Add some more nit changes to this PR.

rt4914 commented on app/src/main/java/org/oppia/app/player/state/StateFragmentPresenter.kt:
```diff
@@ -15,15 +21,70 @@ class StateFragmentPresenter @Inject constructor(
+    llRoot!!.addView(contentComponent, params)
+  }
+
+  /** The function for adding [NumberInputInteractionView]*/
```
Ditto

nikitamarysolomanpvt replied:
resolved

nikitamarysolomanpvt committed:
#190 review fixes

nikitamarysolomanpvt committed:
#190 review fixes

nikitamarysolomanpvt committed:
Merge branches 'develop' and 'exploration-player-3-interaction' of https://github.com/oppia/oppia-android into exploration-player-3-interaction-input-type

nikitamarysolomanpvt committed:
build.gradle revert to develop code

nikitamarysolomanpvt committed:
styles revert to develop code

nikitamarysolomanpvt committed:
End with self-closing tag. /> fix in cellular_data_dialog

nikitamarysolomanpvt committed:
HomeFragment revert to develop code

nikitamarysolomanpvt committed:
files revert to develop code

nikitamarysolomanpvt committed:
files revert to develop code

nikitamarysolomanpvt committed:
files revert to develop code

nikitamarysolomanpvt committed:
files revert to develop code

nikitamarysolomanpvt committed:
files revert to develop code

nikitamarysolomanpvt committed:
files revert to develop code

nikitamarysolomanpvt committed:
files revert to develop code

nikitamarysolomanpvt committed:
files revert to develop code

nikitamarysolomanpvt committed:
files revert to develop code

nikitamarysolomanpvt committed:
files revert to develop code

nikitamarysolomanpvt committed:
files revert to develop code

nikitamarysolomanpvt committed:
files revert to develop code

nikitamarysolomanpvt committed:
files revert to develop code

rt4914 commented on app/src/main/java/org/oppia/app/player/state/StateFragmentPresenter.kt:
```diff
@@ -15,15 +21,70 @@ class StateFragmentPresenter @Inject constructor(
+    llRoot = binding.root.findViewById(R.id.llRoot)
+    addTextInputContentCard("text", 1)
+    addNumberInputContentCard("number 1", 2)
+    addFractionInputContentCard("fraction 1/1", 3)
```
Still not sure what does this 1,2,3 does?

nikitamarysolomanpvt replied:
Removed

rt4914 commented on app/src/main/java/org/oppia/app/player/state/StateFragmentPresenter.kt:
```diff
@@ -15,15 +21,70 @@ class StateFragmentPresenter @Inject constructor(
     return binding.root
   }
 
+  /** The function for adding [TextInputInteractionView] */
```
Expected `.` at the end of sentence.

nikitamarysolomanpvt replied:
resolved

rt4914 commented on app/src/main/java/org/oppia/app/player/state/StateFragmentPresenter.kt:
```diff
@@ -15,15 +21,70 @@ class StateFragmentPresenter @Inject constructor(
+    llRoot!!.addView(contentComponent, params)
+  }
+
+  /** The function for adding [NumberInputInteractionView] */
```
Expected `.` at the end of sentence.

nikitamarysolomanpvt replied:
resolved

rt4914 commented on app/src/main/java/org/oppia/app/player/state/StateFragmentPresenter.kt:
```diff
@@ -15,15 +21,70 @@ class StateFragmentPresenter @Inject constructor(
+
+  }
+
+  /** The function for adding [FractionInputInteractionView] */
```
Expected `.` at the end of sentence.

nikitamarysolomanpvt replied:
resolved

rt4914 commented on app/src/sharedTest/java/org/oppia/app/player/state/StateFragmentPresenterTest.kt:
```diff
@@ -0,0 +1,89 @@
+
+@RunWith(AndroidJUnit4::class)
+class StateFragmentPresenterTest {
+  // The function to test text,number,and fraction input interaction views
```
Nit: Update this comment
`// The function to test text, number and fraction input interaction views.`

nikitamarysolomanpvt replied:
updated

rt4914 reviewed (commented):
PTAL. Nit Changes. Overall this PR looks good to me. I will make a deeper review soon.

nikitamarysolomanpvt committed:
changed double quote to single quotes in build.gradle

nikitamarysolomanpvt committed:
KDoc update in state presenter

veena14cs commented on app/src/main/res/layout/exploration_activity.xml:
```diff
@@ -5,4 +5,4 @@
   android:layout_width="match_parent"
   android:layout_height="match_parent"
-  tools:context=".player.exploration.ExplorationActivity" />
+  tools:context=".player.exploration.ExplorationActivity" />
```
Expected EOF

nikitamarysolomanpvt replied:
resolved

veena14cs commented on app/src/sharedTest/java/org/oppia/app/player/state/StateFragmentPresenterTest.kt:
```diff
@@ -0,0 +1,89 @@
+class StateFragmentPresenterTest {
+  // The function to test text,number,and fraction inputs in interaction views.
+  @Test
+  fun addTextInputContentCard() {
```
please update these test names to include both the condition being tested and the expected outcome. See this [slide](https://docs.google.com/presentation/d/1jPyrVafvlCCaT0qY2r1Z1PYKL6MaklXo2QaO5FiJSKo/edit#slide=id.g62b813ee8d_0_134) for more context.

nikitamarysolomanpvt replied:
Updated

rt4914 reviewed (approved):
LGTM

rt4914 commented:
Resolve merge conflicts

nikitamarysolomanpvt committed:
files revert to develop code

nikitamarysolomanpvt committed:
files revert to develop code

nikitamarysolomanpvt committed:
Merge branches 'develop' and 'exploration-player-3-interaction' of https://github.com/oppia/oppia-android into exploration-player-3-interaction-input-type

# Conflicts:
#	app/src/main/java/org/oppia/app/player/state/StateFragment.kt
#	app/src/main/java/org/oppia/app/player/state/StateFragmentPresenter.kt
#	app/src/main/res/values/colors.xml
#	app/src/main/res/values/dimens.xml

veena14cs reviewed (approved):
LGTM

rt4914 commented on app/src/sharedTest/java/org/oppia/app/player/state/StateFragmentPresenterTest.kt:
```diff
@@ -0,0 +1,89 @@
+  // The function to test text,number,and fraction inputs in interaction views.
+  @Test
+  fun testTextInputContentCard_displaySoftInputAccordingToInputType_userTypeInput() {
+    ActivityScenario.launch(HomeActivity::class.java).use {
```
First, remove pause. We don't need this pause as a developer.

Also, break down this test case in multiple test-cases to check TextInput, FractionInput and NumberInput separately.

rt4914 reviewed (commented):
Changes in test-cases

nikitamarysolomanpvt committed:
statefragment testcase updated as per review

nikitamarysolomanpvt committed:
statefragment testcase updated as per review

nikitamarysolomanpvt committed:
statefragment testcase updated as per review

rt4914 commented:
@nikitamarysolomanpvt If you are not using this PR anymore, please deassign and close this PR.

nikitamarysolomanpvt commented:
#263 is Duplicate PR of this
